Phase4 user guide update

philipanda opened the pull request:
This PR proposes AEM installation instructions which don't give the precise versions of the required packages. It was decided to replace the instructions with giving the precise versions to make sure it works. The PR is mostly duplicated at https://github.com/TrenchBoot/TrenchBoot.github.io/pull/56 with the said changes included. This PR stays as a draft for reference in the future, should a CI system for keeping RPMs up to date be developed.

philipanda committed:
install_aem: Remove lack of UEFI support disclaimer

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
install_aem: Update instructions for Legacy & UEFI support

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

krystian-hebel reviewed (approved):
Changes look good, but we may have broken legacy boot on Intel, and because of that we may need a different disclaimer :disappointed: I'm marking this as approved, but let's put the merging on hold until we have more info or issue to link to.

krystian-hebel commented on docs/user-docs/install_aem.md:
```diff
@@ -93,18 +107,45 @@ sudo qubes-dom0-update --disablerepo="*" --enablerepo=aem --action=reinstall \
+    xen-runtime \
+    grub2-common \
+    grub2-efi-x64 \
+    grub2-efi-x64-modules \
```
I'm getting `No match for argument` for this package, it should be installed, not reinstalled.

krystian-hebel commented on docs/user-docs/install_aem.md:
````diff
@@ -93,18 +107,45 @@ sudo qubes-dom0-update --disablerepo="*" --enablerepo=aem --action=reinstall \
+
+```bash
+sudo qubes-dom0-update --disablerepo="*" --enablerepo=aem --action=reinstall \
+    python3-xen \
+    xen \
+    xen-hypervisor \
+    xen-libs \
+    xen-licenses \
+    xen-runtime \
````
All of the Xen packages report `not available` for reinstall, and are skipped for install. They need `--action=upgrade`, at least with freshly installed R4.2.3. I wonder why it works in openQA, unless it doesn't :thinking:

krystian-hebel replied:
`http://<openQA-IP>/tests/3898/logfile?filename=serial_terminal.txt` - it seems to be properly installed there. Maybe the difference is that package version is included in the command, but since it will change in the future, specifying it here in the documentation would add maintenance effort for each release. Maybe a combination of all three commands (install, upgrade, reinstall) in sequence would work, even though it wouldn't be optimal.

philipanda replied:
>  Maybe the difference is that package version is included in the command, but since it will change in the future, specifying it here in the documentation would add maintenance effort for each release

That's the exact reason why I've removed it in the first place, but I didn't expect that would change how the command works entirely. Wouldn't a sequence of install+reinstall be enough?

krystian-hebel replied:
No, install tries to install package only if it isn't already installed, and reinstall tries to reinstall in the exact same version, in this case failing because there is no such version on `aem` repo.

Now that I think of it, perhaps we should somehow test how it would behave if installed package is newer than what is in `aem`. If I understand those correctly, install won't do anything, because package is already installed, upgrade won't do anything, because installed version is newer than that from repo, and reinstall won't be able to reinstall because `aem` doesn't have package in expected version. Zarhus team planned to do some CI/CD that would automatically create new versions of packages when needed, it could be a valid solution for this problem. @pietrushnic are there any updates on that front?

philipanda replied:
If the process of installing the packages is so complex and error-prone, maybe it would be better to create some kind of a script that does that for you? Such script could check what packages are already installed and decide whether a package has to be installed, upgraded, downgraded or reinstalled.

eg. `qubes-dom0-update --disablerepo='*' --enablerepo=aem --action=list xen-hypervisor |& grep xen-hypervisor | tr -s ' ' | cut -d' ' -f2` outputs the version, like `2001:4.17.5-6.fc37`, or nothing if it is not installed.

pietrushnic replied:
> (...)  Zarhus team planned to do some CI/CD that would automatically create new versions of packages when needed; it could be a valid solution for this problem. @pietrushnic are there any updates on that front?

Yes, we have to get back to that. @macpijan cc

philipanda replied:
Creating a fool-proof sequence of commands when the packages need to be installed, reinstalled, updated and downgraded depending on the situation might be overly complicated. What about a description like that?
https://github.com/TrenchBoot/TrenchBoot.github.io/compare/7b45019057df402ba572ea222644dfab055b7cbf%5E..7b45019057df402ba572ea222644dfab055b7cbf

krystian-hebel replied:
After discussing the matter with Beata, we decided to copy the instructions from blog post with the exact versions for now, and then switch to just the names after the packages are automatically updated. So, I ask you to:

1. Address other comments to have a clean-ish state
2. Mark this PR as draft so it can be reused after CI for keeping RPMs up to date is working
3. Create a new PR with most of the changes from here (without https://github.com/TrenchBoot/TrenchBoot.github.io/compare/7b45019057df402ba572ea222644dfab055b7cbf%5E..7b45019057df402ba572ea222644dfab055b7cbf), but with exact package versions, as in blog post

philipanda replied:
@krystian-hebel There are no package versions given in the blog post. Do you mean the versions used on OpenQA?

krystian-hebel replied:
Sorry, I wanted to add a link here but got distracted. I meant the new blog post, currently in review: https://github.com/3mdeb/news-and-ideas/pull/660/files

philipanda replied:
https://github.com/TrenchBoot/TrenchBoot.github.io/pull/56

philipanda replied:
Oh, your comment did not appear until I've refreshed the page, sorry. Let me verify that it complies with the blog post

krystian-hebel replied:
> #56

It may be worth to edit the description of this PR to include link to #56 as well as explanation why this is a draft and when it should be resurrected. After that, let's move further discussion to #56 as the remaining point from this thread doesn't apply here, but there.

philipanda replied:
Updated the description

krystian-hebel commented on docs/user-docs/install_aem.md:
````diff
@@ -93,18 +107,45 @@ sudo qubes-dom0-update --disablerepo="*" --enablerepo=aem --action=reinstall \
+To check on which partition is your OS installed run:
+
+```bash
+df --output=source /
````
Won't it show `/dev/mapper/luks-*` under Qubes OS?

philipanda replied:
It worked fine when I was writing this. It is not impossible that the disk was not encrypted on that device though. I can't remember.

krystian-hebel replied:
![image](https://github.com/user-attachments/assets/b2ee9e97-1ef3-4b12-b6af-06331776b597)

Works fine for `/boot`, but for `/` the easiest way is to analyze output of `lsblk`.

philipanda replied:
I've found a way to do that by analyzing LVM volumes, which in the end also requires looking up `lsblk`.
[95eb81f](https://github.com/TrenchBoot/TrenchBoot.github.io/pull/53/commits/95eb81f49b7cf5727e68ca8fd400a0e95ae03e99)
Maybe it's bettet to just tell to look up to which device is the `/` LV attached to using lsblk?

krystian-hebel replied:
AEM is pretty much useless without encryption, so I'd remove this case to avoid confusion.

philipanda replied:
https://github.com/TrenchBoot/TrenchBoot.github.io/compare/ef0f39d87829b2d1f89119e6decf6330233733ab%5E..ef0f39d87829b2d1f89119e6decf6330233733ab?expand=1

philipanda committed:
install_aem.md: Add how to find out root disk with LVM+LUKS

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

philipanda committed:
install_aem.md: Installing dependencies: Describe using different --action

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

krystian-hebel commented on docs/user-docs/install_aem.md:
```diff
@@ -93,18 +107,142 @@ sudo qubes-dom0-update --disablerepo="*" --enablerepo=aem --action=reinstall \
+one. `reinstall` won't work because the versions don't match and `install` won't
+work because the package is already installed.
+
+Make sure that all of the packages listed above are installed int the system
```
```suggestion
Make sure that all of the packages listed above are installed in the system
```

krystian-hebel replied:
@philipanda you've missed this one, it is not the same as https://github.com/TrenchBoot/TrenchBoot.github.io/pull/53#discussion_r2131980327 even though this is an exact copy of that text. It should get outdated after the CI is implemented, but just in case let's fix it so whoever, whenever continues this work won't have to do this.

philipanda replied:
sorry https://github.com/TrenchBoot/TrenchBoot.github.io/compare/f15cede8a4945b7b6618ac5c120913a44ca496af..b324773f4ca6756d594e6ddd2a20e6153d4284e3
No need to rebase the #56 as the whole paragraph was replaced there

krystian-hebel commented on docs/user-docs/install_aem.md:
```diff
@@ -93,18 +107,142 @@ sudo qubes-dom0-update --disablerepo="*" --enablerepo=aem --action=reinstall \
+one. `reinstall` won't work because the versions don't match and `install` won't
+work because the package is already installed.
+
+Make sure that all of the packages listed above are installed int the system
```
```suggestion
Make sure that all of the packages listed above are installed in the system
```

philipanda replied:
https://github.com/TrenchBoot/TrenchBoot.github.io/compare/ef0f39d87829b2d1f89119e6decf6330233733ab..f15cede8a4945b7b6618ac5c120913a44ca496af

philipanda committed:
user-docs/install_aem.md: Remove unencrypted drive instructions

Instructions for reinstalling grub2 on Legacy systems.
AEM without encryption is pretty much useless anyway.

Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>
